nemo/collections/llm/gpt/model/hyena.py

```diff
@@ -0,0 +1,698 @@
+    ffn_hidden_size: int = 11008
+    gated_linear_unit: bool = True
+    num_attention_heads: int = 32
+    use_cpu_initialization: bool = False
+    hidden_dropout: float = 0.0
+    attention_dropout: float = 0.0
+    params_dtype: torch.dtype = torch.bfloat16
+    normalization: str = "RMSNorm"
+    add_qkv_bias: bool = False
+    add_bias_linear: bool = False
+    layernorm_epsilon: float = 1e-6
+    recompute_granularity: str = 'full'
+    recompute_method: str = 'uniform'
+    recompute_num_layers: int = 4
+    hyena_init_method: str = 'small_init'
+    hyena_output_layer_init_method: str = 'wang_init'
+    hyena_filter_no_wd: bool = True
+
+
+@dataclass
+class HyenaNV7bConfig(Hyena7bConfig):
+    """
+    Several unintentional design choices were made to the original Arc implementation that are required to use the
+    original Arc checkpoints, but may result in less stable model training. If you are training from scratch,
+    these are the recommended configs.
+    """
+
+    remove_activation_post_first_layer: bool = False
+    add_attn_proj_bias: bool = False
+
+
+@dataclass
+class Hyena40bConfig(HyenaConfig):
+    """Config matching the 40b 8k context Evo2 model"""
+
+    hybrid_override_pattern: str = "SDH*SDHSDH*SDHSDH*SDHSDH*SDHSDH*SDH*SDHSDH*SDHSDH*"
+    num_layers: int = 50
+    seq_length: int = 8192
+    hidden_size: int = 8192
+    num_groups_hyena: int = 8192
+    num_groups_hyena_medium: int = 512
+    num_groups_hyena_short: int = 512
+    make_vocab_size_divisible_by: int = 8
+    tokenizer_library: str = 'byte-level'
+    mapping_type: str = "base"
+    ffn_hidden_size: int = 21888
+    gated_linear_unit: bool = True
+    num_attention_heads: int = 64
+    use_cpu_initialization: bool = False
+    hidden_dropout: float = 0.0
+    attention_dropout: float = 0.0
+    params_dtype: torch.dtype = torch.bfloat16
+    normalization: str = "RMSNorm"
+    add_qkv_bias: bool = False
+    add_bias_linear: bool = False
+    layernorm_epsilon: float = 1e-6
+    recompute_granularity: str = 'full'
+    recompute_method: str = 'uniform'
+    recompute_num_layers: int = 2
+    hyena_init_method: str = 'small_init'
+    hyena_output_layer_init_method: str = 'wang_init'
+    hyena_filter_no_wd: bool = True
+
+
+@dataclass
+class HyenaNV40bConfig(Hyena40bConfig):
+    """
+    Several unintentional design choices were made to the original Arc implementation that are required to use the
+    original Arc checkpoints, but may result in less stable model training. If you are training from scratch,
+    these are the recommended configs.
+    """
+
+    remove_activation_post_first_layer: bool = False
+    add_attn_proj_bias: bool = False
+
+
+@dataclass
+class Hyena7bARCLongContextConfig(Hyena7bConfig):
+    """The checkpoint from ARC requires padding to the FFN dim
+    due to constraintes from large TP size for training."""
+
+    ffn_hidden_size: int = 11264
+
+
+@dataclass
+class Hyena40bARCLongContextConfig(Hyena40bConfig):
```

Comment:
Up to you but should we move these config classes up before the importer?

Reply:
Good idea, I'll do that next week.